src/lighteval/metrics/metrics_corpus.py

```diff
@@ -103,7 +103,7 @@ def __init__(self, metric_type: str):
     def compute(self, items: list[GenerativeCorpusMetricInput]) -> float:
         """Computes the metric score over all the corpus generated items, by using the sacrebleu implementation."""
         golds = [i.golds for i in items]
-        preds = [as_list(i.preds) for i in items]
+        preds = [pred for i in items for pred in i.preds]
```

Comment:
If we have several predictions, this might lead to having different length of streams between gold and preds. It might be better to 1. display a warning if we have more than one pred and 2. only take the first prediction of each prediction set instead of flattening everything

Reply:
Good point, thanks. Changed it accordingly.